diff --git a/lib/ata.js b/lib/ata.js
index e0a03eb1b4d923e66fb92405eced8b618074ce02..272e905d50ca2da0d777b4f78a9200b481ee5811 100644
--- a/lib/ata.js
+++ b/lib/ata.js
@@ -32,21 +32,21 @@ export function activateAutomaticTypeAcquisition(env, onFetch) {
                     };
                 }
                 if (fileName.startsWith('/node_modules/')) {
-                    const path = uri.substring('/node_modules/'.length);
+                    const path = fileName.substring('/node_modules/'.length);
                     return await _stat(path);
                 }
             },
             async readFile(uri) {
                 const fileName = env.typescript.uriToFileName(uri);
                 if (fileName.startsWith('/node_modules/')) {
-                    const path = uri.substring('/node_modules/'.length);
+                    const path = fileName.substring('/node_modules/'.length);
                     return await _readFile(path);
                 }
             },
             async readDirectory(uri) {
                 const fileName = env.typescript.uriToFileName(uri);
                 if (fileName.startsWith('/node_modules/')) {
-                    const path = uri.substring('/node_modules/'.length);
+                    const path = fileName.substring('/node_modules/'.length);
                     return _readDirectory(path);
                 }
                 return [];
@@ -185,10 +185,10 @@ export function activateAutomaticTypeAcquisition(env, onFetch) {
         const parts = path.split('/');
         let pkgName = parts[0];
         if (pkgName.startsWith('@')) {
-            if (parts.length < 2 || !parts[1]) {
+            if (!parts[1]) {
                 return undefined;
             }
-            pkgName += '/' + parts[2];
+            pkgName += '/' + parts[1];
         }
         return pkgName;
     }
